polynote-frontend/polynote/cell.js

```diff
@@ -407,6 +423,52 @@ export class CodeCell extends Cell {
     get content() {
         return this.editor.getValue();
     }
+
+    applyEdits(edits) {
+        // can't listen to these edits or they'll be sent to the server again
+        // TODO: is there a better way to silently apply these edits? This seems like a hack; only works because of
+        //       single-threaded JS, which I don't know whether workers impact that assumption (JS)
+        this.applyingServerEdits = true;
+
+        try {
+            const monacoEdits = [];
+            const model = this.editor.getModel();
+            edits.forEach((edit) => {
+                if (edit.isEmpty()) {
+                    return;
```

Comment:
Is this just defensive or is it possible that an edit would be empty?

Reply:
I think I've weeded out the places where empty edits get generated. So I guess it's defensive at this point. That said, empty edits can cause weird problems if they make it to monaco (at least it seemed that way)

Reply:
Hmm, maybe we should at least log these situations so if things go wonky we might have some clues?